backend/sample.js

```diff
@@ -1,120 +1,286 @@
+      const folderToAdd = new Folder(req.body);
+      folderToAdd.save()
+      res.status(200).send(folderToAdd).end();
     } else {
-        if (!result["isPrivate"]) {
-            res.status(201).send(result);
-        } else {
-            if (passw["password"] === result["password"]) {
-                res.status(201).send(result);
-            } else {
-                //res.status(404).send(req.body);
-                res.status(404).send("Wrong password. Access denied.");
-            }
-        }
+      res.status(404).send("Duplicate file name. "+isDup).end();
     }
-});
-//open note
-app.post("/:folderName/:note", (req, res) => {
-    const fName = req.params["folderName"];
-    const noteToGet = req.params["note"];
-    const result = findFolder(fName);
+  }
+  else if(keyword===undefined||keyword.length==0){
+    let result = await findFolder(folder);
     if (result === undefined || result.length == 0) {
-        res.status(404).send("Folder not found.");
+      res.status(404).send("Folder not found.");
     } else {
-        result = findNote(fName, noteToGet);
-        if (result === undefined || result.length == 0) {
-            res.status(404).send("Note not found.");
+      if (!result["isPrivate"]) {
+        res.status(200).send(result);
+      } else {
+        let passw = req.body["password"];
+        if (passw["password"] === result["password"]) {
+          //res.status(200).send(result);
+          res.status(200).send("get notes");
+          //res.status(200).send(findNotesByFolder(folderName));
         } else {
-            if (!result["isPrivate"]) {
-                res.status(201).send(result).end();
-            } else {
-                const passw = req.body;
-                if (passw["password"] === result["password"]) {
-                    res.status(201).send(result).end();
-                } else {
-                    res.status(404).send("Wrong password. Access denied.");
-                }
-            }
+          //res.status(404).send(req.body);
+          res.status(404).send("Wrong password. Access denied.");
         }
+      }
     }
-});
+  }
+  else{
+    //search folder
+    Folder.find({"name":{"$regex": keyword, "$options":"i"}}).exec(function (err, docs) {
+      if (err){
+          console.log(err);
+      }else{
+        res.status(200).send(docs).end();
+      }});
+    //res.status(200).send(result).end();
+  }
+})
+function searchFolder(keyword){
 
-//search note
-app.get("/:folderName", (req, res) => {});
-//add folder
-app.post("/", (req, res) => {
-    const { name, color, isPrivate, password } = req.body;
-    isDup = findFolder(name);
-    if (true) {
-        const folderToAdd = new Folder({ name, color, isPrivate, password });
-        folderToAdd.save();
-        res.status(201).send(folderToAdd).end();
+}
+/*add folder
+ Comment: These two are merged into one post function (see above). But I keep them for now just in case.
```

Comment:
are the new functions working or is this just a temporary comment?

Reply:
the new functions were working but after moving async functions to connections.js they fail. I'll keep working on them to resolve the problem.